m/method_source/method_source_ubi_8.5.sh

```diff
@@ -1,42 +1,74 @@
-# Tested on     : UBI: 8.5
-# Language      : Ruby
-# Travis-Check  : True
-# Script License: Apache License 2.0
-# Maintainer's  : Balavva Mirji <Balavva.Mirji@ibm.com>
+# Package               : method_source
+# Version               : v0.8.2,v0.9.0,v1.0.0
+# Source repo           : https://github.com/banister/method_source
+# Tested on             : UBI 8.5
+# Language              : Ruby
+# Travis-Check          : True
+# Script License        : Apache License, Version 2 or later
+# Maintainer            : Vathsala . <vaths367@in.ibm.com>
 #
-#
-# Disclaimer: This script has been tested in root mode on given
-# ==========  platform using the mentioned version of the package.
-#             It may not work as expected with newer versions of the
-#             package and/or distribution. In such case, please
-#             contact "Maintainer" of this script.
+# Disclaimer            : This script has been tested in root mode on given
+# ==========              platform using the mentioned version of the package.
+#                         It may not work as expected with newer versions of the
+#                         package and/or distribution. In such case, please
+#                         contact "Maintainer" of this script.
 #
 # ----------------------------------------------------------------------------
 
+export PACKAGE_NAME=method_source
+export PACKAGE_URL=https://github.com/banister/method_source
+
+if [ -z "$1" ]; then
+  export PACKAGE_VERSION=${1:-v1.0.0}
+else
+  export PACKAGE_VERSION=$1
+fi
+if [ -d "${method_source}" ] ; then
+  rm -rf ${method_source}
+fi
+
 
-set -e
+yum install -y git ruby procps yum-utils wget
 
-PACKAGE_NAME=method_source
-PACKAGE_VERSION=${1:-v0.9.2}            
-PACKAGE_URL=https://github.com/banister/method_source
+gem install bundle 
+gem install rake 
+curl -sSL https://rvm.io/mpapis.asc | gpg2 --import - 
+curl -sSL https://rvm.io/pkuczynski.asc | gpg2 --import - 
+curl -L https://get.rvm.io | bash -s stable 
+source /etc/profile.d/rvm.sh
+rvm install ruby-2.7
+gem install bundler:1.17.3
+gem install kramdown-parser-gfm
 
-yum install git ruby ruby-devel -y
-gem install bundle
+git clone ${PACKAGE_URL} ${PACKAGE_NAME}
+cd ${PACKAGE_NAME}
+git checkout ${PACKAGE_VERSION}
+ret=$?
+if [ $ret -eq 0 ] ; then
+ echo "$PACKAGE_VERSION found to checkout "
+else
+ echo "$PACKAGE_VERSION not found "
+ exit
+fi
 
-git clone $PACKAGE_URL
-cd $PACKAGE_NAME
-git checkout $PACKAGE_VERSION
+#Build and test
 
-export BUNDLE_GEMFILE=$PWD/Gemfile
+bundle install
 
-which bundle || gem install bundler
-gem update bundler
+ret=$?
+if [ $ret -ne 0 ] ; then
+  echo "Build failed "
+else
+  bundle exec rake
+  ret=$?
+  if [ $ret -ne 0 ] ; then
+    echo "Tests failed "
+  else
+    echo "Build & unit tests Successful "
+  fi
+fi
 
-bundle install --verbose
-bundle exec rake
\ No newline at end of file
+#Observed one failure:syntax error  and is in parity with Intel
```

Comment:
Please specify the test failure details, syntax error seems like it needs to be looked at, does the build work for top-of-tree?

Reply:
Added detailed error log

Reply:
Travis check passed..build and test both passing,deleted the failure comment